### ISSUE TITLE
New SDK request: Golang SDK for DropboxSign

### ISSUE THREAD

**Issue by abhishekdas29** [hellosign/hellosign-openapi#313]
Hello Team,

Do you have any plans for supporting the golang SDK generated from the OpenAPI specs?

I have a fork of the repo and have a draft PR to have it generated. I don't know if this is something which you want to bring in.
https://github.com/abhishekdas29/hellosign-openapi/pull/1

**Comment by jtreminio-dropbox** [comment]
Unfortunately one of the hacky fixes we were forced to add to our OpenAPI spec was setting all endpoints as `application/json` first, and `multipart/form-data` second (if needed).

This is due to all the other generators creating completely different code depending on the request content type (`multipart/form-data` creating API methods where all arguments are listed, vs a single typed object).

This does not appear to be the case for the Golang generator, but I don't see that it correctly switches to `multipart/form-data` when a file upload is detected.

* [Here is what we did to the Node files](https://github.com/hellosign/hellosign-openapi/blob/49edd326b31a883d0b9f7783ea1415f9b27974f8/sdks/node/api/apis.ts#L52)
* [Here is what we did to the PHP files](https://github.com/hellosign/hellosign-openapi/blob/3d07e69800d7fc74c205c423233fd9c2395c03e2/sdks/php/src/ObjectSerializer.php#L478)

We would need to apply the same logic to this new Golang SDK to consider it.

**Comment by abhishekdas29** [comment]
@jtreminio-dropbox - So are you saying that the auto generated files were manually overwritten in the PHP and node sdk?

**Comment by jtreminio-dropbox** [comment]
Yes, all of our currently supported SDKs had their mustache templates changed to support what I listed above.
